AsyncDisplayKit/Details/ASCollectionViewFlowLayoutInspector.m

```diff
@@ -67,11 +63,7 @@ - (void)didChangeCollectionViewDelegate:(id<ASCollectionDelegate>)delegate;
 
 - (void)didChangeCollectionViewDataSource:(id<ASCollectionDataSource>)dataSource
 {
-  if (dataSource == nil) {
-    memset(&_dataSourceFlags, 0, sizeof(_dataSourceFlags));
-  } else {
-    _dataSourceFlags.implementsNumberOfSectionsInCollectionView = [dataSource respondsToSelector:@selector(numberOfSectionsInCollectionView:)];
-  }
+  // nop
```

Comment:
Good point! I did that, and added `AS_SUBCLASSING_RESTRICTED` to this class.

This is a breaking change! But subclassing this was never intended in the first place, and now we've just made it formal.